flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/common/CommonExecSink.java

```diff
@@ -225,15 +226,19 @@ public DynamicTableSinkSpec getTableSinkSpec() {
     /**
      * Returns a List of Tuples, each Tuple holds 2 integers, the char field idx and its precision.
      */
-    private List<Tuple2<Integer, Integer>> getCharFieldIndices(RowType physicalType) {
-        final List<Tuple2<Integer, Integer>> charFieldsAndLengths = new ArrayList<>();
+    private List<Tuple3<Integer, Integer, Boolean>> getCharFieldIndices(RowType physicalType) {
+        final List<Tuple3<Integer, Integer, Boolean>> charFieldsAndLengths = new ArrayList<>();
```

Comment:
Tuples are not very convenient for code readability. Also the method name `getCharFieldIndices` is not useful for what the function is doing. Can't we introduce a little POJO with default scoped members and an assigning constructor?